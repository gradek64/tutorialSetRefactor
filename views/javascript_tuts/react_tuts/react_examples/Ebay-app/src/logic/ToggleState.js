import React from 'react';
import PropTypes from 'prop-types';

/*
  *@this is very clever class all it does it provides logic for toggling (state of true or false)
  and expose it toggleMethod and toggle state in props.render to other components
  *@
*/
class ToggleState extends React.Component {
  state = {
    toggle: false,
  };

  toggle = (toggle) => {
    this.setState({ toggle: !toggle });
  };

  render() {
    const { toggle } = this.state;
    const { renderComp } = this.props;
    return (
      <React.Fragment>{renderComp(this.toggle, toggle)}</React.Fragment>
    );
  }
}

ToggleState.propTypes = {
  renderComp: PropTypes.func.isRequired,
};

export default ToggleState;
