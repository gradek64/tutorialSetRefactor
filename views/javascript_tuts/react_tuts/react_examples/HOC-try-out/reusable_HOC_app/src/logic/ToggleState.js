import React from 'react';
import PropTypes from 'prop-types';

/*
  *@this is very clever class all it does it provides logic for toggling (state of true or false)
  and expose it toggleMethod and toggle state in props.render to other components
  *@
*/
class ToggleState extends React.Component {
  state = {
    toggle: false,
  };
  toggle = (toggle) => {
    this.setState({ toggle: !toggle });
  };

  render() {
    const { toggle } = this.state;
    return (
      <React.Fragment>{this.props.render(this.toggle, toggle)}</React.Fragment>
    );
  }
}

ToggleState.propTypes = {
  render: PropTypes.func.isRequired,
};

export default ToggleState;
