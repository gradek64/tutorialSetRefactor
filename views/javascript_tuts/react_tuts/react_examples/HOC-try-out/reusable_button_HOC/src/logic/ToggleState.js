import React from "react";
import PropTypes from "prop-types";

class ToggleState extends React.Component {
  state = {
    toggle: false
  };
  toggle = toggle => {
    this.setState({ toggle: !toggle });
  };

  render() {
    const { toggle } = this.state;
    return (
      <React.Fragment>{this.props.render(this.toggle, toggle)}</React.Fragment>
    );
  }
}

ToggleState.propTypes = {
  render: PropTypes.func.isRequired
};

export default ToggleState;
