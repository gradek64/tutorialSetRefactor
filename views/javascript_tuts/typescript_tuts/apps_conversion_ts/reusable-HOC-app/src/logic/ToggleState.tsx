import * as React from 'react';

/*
  *@this is very clever class all it does it provides logic for toggling (state of true or false)
  and expose it toggleMethod and toggle state in props.render to other components
  *@
*/

//class itself
interface IToggleState {
  toggle:(toggle:Boolean)=>void;
}

//react class interface
interface IProps {
  render:(
    toggleMethod:(toggle:Boolean)=>void,
    toggleState:Boolean
    )=>{};
}
interface IState {
  toggle:Boolean;
}

class ToggleState extends React.Component<IProps,IState> implements IToggleState {
  state:IState = {
    toggle: false,
  };
  toggle = (toggle:Boolean) => {
    this.setState({ toggle: !toggle });
  };

  render() {
    const { toggle } = this.state;
    const { render } = this.props;
    return (
      <React.Fragment>{render(this.toggle, toggle)}</React.Fragment>
    );
  }
}


export default ToggleState;
