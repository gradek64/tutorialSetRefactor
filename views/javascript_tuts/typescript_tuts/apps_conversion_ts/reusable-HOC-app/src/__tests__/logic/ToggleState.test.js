import React from 'react';
import { shallow } from 'enzyme';
import { noReactFragementSupport } from '../../test_helpers';

/*
  *@This class instance and the way enzyme is getting methods and properties is by
  *@    (class) means wrapper.instance();
*/

/*
  *@this next line is defined in a root/jest.configuration.js
  *@so no need to import library
*/
// import toJson from 'enzyme-to-json';

// component for snapshot test;
import ToggleState from '../../logic/ToggleState';

let wrapper;
beforeAll(() => {
  wrapper = shallow(<ToggleState render={() => {}} />);
});

test('It renders React.Fragement', () => {
  wrapper = shallow(<ToggleState render={() => {}} />);
  /*
    *@Jest doent seems to render <React.Fragement>
    *@U need to utility method to swap render outcome
  */
  const wrapperUpdate = noReactFragementSupport(wrapper.debug());
  expect(wrapperUpdate).toMatchSnapshot();
});
test('toggleStates toggle from false to true and vice-versa', () => {
  const initialToggleState = wrapper.instance().state.toggle;

  // call class toggle
  wrapper.instance().toggle(initialToggleState);

  const updatedToggleStateRead = wrapper.instance().state.toggle;
  expect(updatedToggleStateRead).toEqual(!initialToggleState);

  // and call back toggle;
  wrapper.instance().toggle(updatedToggleStateRead);

  expect(wrapper.instance().state.toggle).toEqual(!updatedToggleStateRead);
});
