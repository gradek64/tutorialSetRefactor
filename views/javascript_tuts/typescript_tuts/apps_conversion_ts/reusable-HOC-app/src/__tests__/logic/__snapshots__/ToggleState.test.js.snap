// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders React.Fragement 1`] = `"<React.Fragment />"`;
